code/modules/projectiles/ammo_datums.dm

```diff
@@ -3340,3 +3340,144 @@
 	var/obj/item/clothing/mask/facehugger/child = new(T)
 	child.hivenumber = hugger_hive
 	INVOKE_ASYNC(child, TYPE_PROC_REF(/obj/item/clothing/mask/facehugger, leap_at_nearest_target))
+
+/*
+//========
+					SHARP Dart Ammo
+//========
+*/
+/datum/ammo/rifle/sharp
+	name = "dart"
+	ping = null //no bounce off.
+	damage_type = BRUTE
+	shrapnel_type = /obj/item/sharp
+	flags_ammo_behavior = AMMO_SPECIAL_EMBED|AMMO_NO_DEFLECT|AMMO_STRIKES_SURFACE_ONLY|AMMO_HITS_TARGET_TURF
+	icon_state = "sonicharpoon"
+	var/embed_object = /obj/item/sharp/explosive
+
+	shrapnel_chance = 100
+	accuracy = HIT_ACCURACY_TIER_MAX
+	accurate_range = 12
+	max_range = 7
+	damage = 35
+	shell_speed = AMMO_SPEED_TIER_2
+
+/datum/ammo/rifle/sharp/on_embed(var/mob/embedded_mob, var/obj/limb/target_organ)
+	if(!ishuman(embedded_mob))
+		return
+	var/mob/living/carbon/human/humano = embedded_mob
+	if(humano.species.flags & NO_SHRAPNEL)
+		return
+	if(istype(target_organ))
+		target_organ.embed(new embed_object)
+
+/datum/ammo/rifle/sharp/on_hit_obj(obj/O, obj/item/projectile/P)
+	drop_dart(P.loc, P)
+
+/datum/ammo/rifle/sharp/on_hit_turf(turf/T, obj/item/projectile/P)
+	drop_dart(T, P)
+
+/datum/ammo/rifle/sharp/do_at_max_range(obj/item/projectile/P)
+	drop_dart(P.loc, P)
+
+/datum/ammo/rifle/sharp/proc/drop_dart(var/loc, obj/item/projectile/P)
+	new embed_object(loc, P.dir)
+
+/datum/ammo/rifle/sharp/explosive
+	name = "9X-E sticky explosive dart"
+
+/datum/ammo/rifle/sharp/explosive/on_hit_mob(mob/living/M, obj/item/projectile/P)
+	if(!M || M == P.firer) return
+	var/mob/shooter = P.firer
+	shake_camera(M, 2, 1)
+	if(shooter && ismob(shooter))
+		if(!M.get_target_lock(shooter.faction_group))
+			var/obj/item/weapon/gun/rifle/sharp/weapon = P.shot_from
+			if(weapon && weapon.explosion_delay_sharp)
+				addtimer(CALLBACK(src, PROC_REF(delayed_explosion), P, M, shooter), 5 SECONDS)
+			else
+				addtimer(CALLBACK(src, PROC_REF(delayed_explosion), P, M, shooter), 1 SECONDS)
+
+/datum/ammo/rifle/sharp/explosive/drop_dart(var/loc, obj/item/projectile/P, var/mob/shooter)
+	var/signal_explosion = FALSE
+	if(locate(/obj/item/explosive/mine) in get_turf(loc)) // TO FIX MID SOLUTION?
+		signal_explosion = TRUE
+	var/obj/item/explosive/mine/sharp/dart = new /obj/item/explosive/mine/sharp(loc)
+	// if no darts on tile, don't arm, explode instead.
+	if(signal_explosion)
+		INVOKE_ASYNC(dart, TYPE_PROC_REF(/obj/item/explosive/mine/sharp, prime), shooter)
+	else
+		dart.anchored = TRUE
+		addtimer(CALLBACK(dart, TYPE_PROC_REF(/obj/item/explosive/mine/sharp, deploy_mine), shooter), 3 SECONDS, TIMER_DELETE_ME)
+		addtimer(CALLBACK(dart, TYPE_PROC_REF(/obj/item/explosive/mine/sharp, disarm)), 1 MINUTES, TIMER_DELETE_ME)
+
+/datum/ammo/rifle/sharp/explosive/proc/delayed_explosion(obj/item/projectile/P, mob/M, mob/shooter)
+	if(istype(M, /mob))
+		var/explosion_size = 100
+		var/falloff_size = 50
+		var/cause_data = create_cause_data("P9 SHARP Rifle", shooter)
```

Comment:
not P.name?

Reply:
No, P can be qdel'ed before due to another blowing up the one currently processing the delayed explosion.

Reply:
ok :3